staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/tlsconfig.go

```diff
@@ -98,22 +98,38 @@ func (c *DynamicServingCertificateController) GetConfigForClient(clientHello *tl
 
 	// if the client set SNI information, just use our "normal" SNI flow
 	if len(clientHello.ServerName) > 0 {
+		klog.Infof("GetConfigForClient: The client specified ServerName = %v, performing normal SNI flow", clientHello.ServerName)
```

Comment:
I think that the normal SNI flow will first check `NameToCertificate` and then `Certificates[0]`